tldr.bats

```diff
@@ -155,6 +155,17 @@ setup() {
   [[ "$output" = *"https://manned.org/7z"* ]]
 }
 
+# Source: https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md#page-structure
+# bats test_tags=required
+@test "REQUIRED: supports escaping the placeholder syntax" {
+  run $PATH_TO_TLDR_CLIENT docker inspect
+  [[ $status -eq 0 ]]
+  [[ "$output" = *"{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}"* ]]
```

Comment:
```suggestion
  [[ "$output" = *"'{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' container`"* ]]
```

Reply:
I don't think that suggestion adds much value, meanwhile it does add potential complexity with handling different styles, colors, or ANSI codes between arguments in different clients.

I think we should stick to the original proposal.

Reply:
Maybe split this into the past version and this suggestion, making the latter optional, with NO_COLOR set to true? I do think that the backslashes can confuse clients.

Reply:
Which test are you referring to there?

I don't think we're testing escaped curly braces yet. It was a relatively recent addition to the spec compared to when I wrote these tests.

Reply:
Sorry, I was talking about the other line of this test

Reply:
These two tests are different.

1. One tests that escaped curly braces are rendered correctly. `\{\{arg\}\}` → `{{arg}}`
2. One ensures that double backslash (normally interpreted as escaped backslashes) are rendered correctly. It does not test escaped curly braces. `\\arg` → `\\arg`

Reply:
Sure, I'm not saying they are equivalent. But the double backslash test, want it or not, also tests the rendering of the placeholders, so it should theoretically suffer from the same problems you mentioned

Reply:
I don't understand what you're trying to say. These are both handling entirely different scenarios. It's possible to craft input that could pass or fail both, but this is about how escapes are handled, which any clients could handle one escape correctly but not the other.

Reply:
See Microsoft documentation:
> mount -o anon \\ServerIP\ShareName Z:

The original Markdown is:

> mount \\{{computer_name}}\{{share_name}} {{Z:}}

If I'm getting it right, it should be rendered as:

> mount \\{{computer_name}}\{{share_name}} {{Z:}}

Yeah, they are different, because the mount page doesn't test escaped placeholders, but it does test text between the placeholders, so it could theorically suffer from the problems you mentioned such as ANSI codes.

I hope you understand me now :sweat_smile:

Reply:
Ahh, I'm following now! Yeah, the second test is actually written wrong then in that case.

> Don't include curly braces in tests, clients are free to remove/replace them.

That really should say placeholder syntax, escaped curly braces are fine. Clients are free to render it as `\\{{computer_name}}\{{share_name}}` or `\\computer_name\share_name`, usually if a client does the latter they use colors/styles to signify the args.

It should only check what it wants to test, so the `\\` part, or could check for multiple tokens like `\\` and `computer_name`, but the assertions should not include the `{{`.